crates/sui-bridge/src/config.rs

```diff
@@ -417,6 +418,43 @@ pub fn read_bridge_client_key(path: &PathBuf) -> Result<SuiKeyPair, anyhow::Erro
         .map_err(|e| anyhow!("Error decoding authority key: {:?}", e))
 }
 
+/// Read a SuiKeyPair from a file. The content could be any of the following:
+/// - Base64 encoded `flag || privkey` for ECDSA key
+/// - Base64 encoded `privkey` for Raw key
+/// - Hex encoded `privkey` for Raw key
+/// If `require_secp256k1` is true, it will return an error if the key is not Secp256k1.
+pub fn read_key(path: &PathBuf, require_secp256k1: bool) -> Result<SuiKeyPair, anyhow::Error> {
+    if !path.exists() {
+        return Err(anyhow::anyhow!("Key file not found at path: {:?}", path));
+    }
+    let file_contents = std::fs::read_to_string(path)?;
+    let contents = file_contents.as_str().trim();
+
+    // Try base64 encoded SuiKeyPair `flag || privkey`
+    if let Ok(key) = SuiKeyPair::decode_base64(contents) {
+        if require_secp256k1 && !matches!(key, SuiKeyPair::Secp256k1(_)) {
```

Comment:
I guess you could use the `if let` pattern for a single match, if you found it more expressive